src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -627,10 +627,25 @@ ErrorIfInsertSelectQueryNotSupported(Query *queryTree, RangeTblEntry *insertRte,
 {
 	Query *subquery = NULL;
 	Oid selectPartitionColumnTableId = InvalidOid;
+	ListCell *rangeTableCell = NULL;
 
 	/* we only do this check for INSERT ... SELECT queries */
 	AssertArg(InsertSelectQuery(queryTree));
 
+	/* we do not expect to see a view in modify target */
+	foreach(rangeTableCell, queryTree->rtable)
+	{
+		RangeTblEntry *rangeTableEntry = (RangeTblEntry *) lfirst(rangeTableCell);
+		if (rangeTableEntry->rtekind == RTE_RELATION &&
+			rangeTableEntry->relkind == RELKIND_VIEW)
+		{
+			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+							errmsg("cannot perform distributed planning for the given "
+								   "modification"),
+							errdetail("Cannot modify views")));
+		}
+	}
```

Comment:
target RTE is determined from Query->resultRelation field. When we are inserting into a view, postgresql uses a pseudo-range table entry that points to the actual relation that would be receiving insert.

Therefore,  checking target RTE to see if it is a view does not provide any useful information. However, range table list contains 2 range table entries of type ```RELKIND_VIEW``` .  They are located at the 1st and 2nd places in range table list.